### PR TITLE
Update rational_best for the herbie eval

### PR DESCRIPTION
**Pull request by oflatt**
This PR simplifies rational_best, which makes herbie perform better
The branch is called "unsound" because the z3 verifier still allows dropping variables (which is unsound)

### CODE REVIEW

File: src/enumo/scheduler.rs [hunk 1]
```diff
@@ -137,7 +137,7 @@ impl Scheduler {
         egraph: &EGraph<L, SynthAnalysis>,
         ruleset: &Ruleset<L>,
     ) -> EGraph<L, SynthAnalysis> {
-        self.run_internal(egraph, ruleset, None, 200_000)
+        self.run_internal(egraph, ruleset, None, 1_000_000)
```
Comment (oflatt) [inline src/enumo/scheduler.rs:140]: reverted this sus change

File: tests/rational.rs [hunk 2]
```diff
@@ -740,26 +817,49 @@ pub mod test {
         let start = Instant::now();
         let best_rules = best_enumo_recipe();
         let duration = start.elapsed();
+        // herbie and ruler can't learn any if rules
+        // but crash if you include them
+        let without_if: Ruleset<Math> = Ruleset::new(
```
Comment (oflatt) [inline tests/rational.rs:816]: Importantly, we have to leave out the div rules

File: src/lib.rs [hunk 3]
```diff
@@ -88,11 +88,10 @@ impl Limits {
         }
     }
 
-    // match oopsla limits
     pub fn deriving() -> Self {
         Self {
             iter: 5,
-            node: 100_000,
```
Comment (ajpal) [inline src/lib.rs:95]: I think it's important that we run with the same limits as Oopsla in order for it to be a fair comparison, I don't think we should change this.

File: tests/rational.rs [hunk 4]
```diff
@@ -793,10 +887,9 @@ pub mod test {
 
         let guarded_rules = run_workload(terms, all_rules.clone(), Limits::rulefinding(), false);
         guarded_rules.to_file("guard.rules");
-        assert!(guarded_rules.0.contains_key("(if ?a 1 (/ 1 ?a)) ==> 1"));
```
Comment (ajpal) [inline tests/rational.rs:796]: Wait, does this mean we don't learn this rule anymore? That's not good.

Comment (oflatt) [reply tests/rational.rs:780]: I made the recipe simpler so it doesn't find if rules like these

File: tests/rational.rs [hunk 5]
```diff
@@ -283,6 +312,53 @@ impl Math {
...
+                    let new_path_neg = z3::ast::Bool::and(&ctx, &[&path, &cond_real._eq(&zero)]);
+                    res.extend(Self::error_conditions(ctx, list[2].clone(), new_path_pos));
+                    res.extend(Self::error_conditions(ctx, list[3].clone(), new_path_neg));
+                } else {
+                    for s in list {
+                        res.extend(Self::error_conditions(ctx, s, path.clone()));
+                    }
+                };
+            }
+            Sexp::String(_atom) => (),
```
Comment (ajpal) [inline tests/rational.rs:355]: nit: _ if unused

File: tests/rational.rs [hunk 6]
```diff
@@ -283,6 +312,53 @@ impl Math {
         }
     }
 
+    fn error_conditions<'a>(
```
Comment (ajpal) [inline tests/rational.rs:315]: can you write a doc comment for this function since it's a little complex?

**Review by ajpal — APPROVED** [review]
lgtm, a few little comments to resolve before merging!
also make sure to merge in main because I merged my cleanup PR so you might need to resolve conflicts

**Comment by oflatt** [comment]
Thanks for the review, fixed it up @ajpal

**Comment by oflatt** [comment]
Spoke too soon, need to fix up a test

**Pull request merged**